### ISSUE TITLE
"TypeError: a is undefined" on Firefox but not Chrome

### ISSUE THREAD

**Issue by chrismurrph** [scala-js/scala-js#1688]
Get the above message in the browser console from Firefox. 
    "org.scala-js" %%% "scalajs-dom" % "0.8.0"
With the 'full' (rather than fast) Javascript generation.
I can setup the site so it can be visited if you like...
Or pop the generated jar file that gives the issue to you...

**Comment by sjrd** [comment]
I'll need more information than that. Do you have a reproduction? What code triggers the error? What version of Scala.js do you use?

The ideal for me is a GitHub repo or a zip of a project that reproduces the problem. If you can't do that, the generated js files for fast and full would be good.

**Comment by chrismurrph** [comment]
If you go to here now on Chrome it will display and tell you that 'Bag Sampling is ON'
http://www.seasoft.com.au:8080/atmosphere/
Same in Firefox.
The error will happen in Firefox if you click in a different spreadsheet cell.
I thought this line was enough for the version:
"org.scala-js" %%% "scalajs-dom" % "0.8.0"

**Comment by lihaoyi** [comment]
@chrismurrph the traditional thing to do in these cases is to minimize the problem. Have you tried doing that?

**Comment by lihaoyi** [comment]
Would be much easier for @sjrd to figure out if you have a self contained example to look at ^_^

**Comment by chrismurrph** [comment]
Hi Li. Yes but I can't do that right now, just doing the best I can on a bit of a deadline. Will get back to this later. To reproduce it ideally I should produce a small spreadsheet on my GitHub that demonstrates the problem. At least you guys are now on the lookout for the bug - just click in another cell to see it.

**Comment by sjrd** [comment]
I just get this on your link:

    Ooops! The server was not able to produce a timely response to your request.
    Please try again in a short while!

**Comment by chrismurrph** [comment]
These are my images for Firefox and Chrome. Shows that Chrome gives a better?? report as does the ERR_INVALID_CHUNKED_ENCODING before the 'simple is not defined'. These from just now - should be same as when you looked. This is with the fast version. 

![firefox](https://cloud.githubusercontent.com/assets/10278575/7958823/6a40b078-0a36-11e5-8271-3df3cf0029a7.png)
![chrome](https://cloud.githubusercontent.com/assets/10278575/7958824/70197250-0a36-11e5-9b39-333f4c79e180.png)

From now on I'm going to try to adjust the setup so permanently have different Page.classes at the two places - external one being full and internal development one being fast. Need to do that to take account of the two different naming conventions used: 
script(src:= "/atmosphere/atmosphere-fastopt.js") 
script(src:= "/atmosphere/atmosphere-opt.js")

**Comment by sjrd** [comment]
The `ERR_INVALID_CHUNKED_ENCODING` error is a problem with your server that has nothing to do with Scala.js. Just fetching the .js file (not even executing it) causes the error, so it just can't be Scala.js.

The `ReferenceError` that follows is a consequence of your browser not being able to fetch the .js file, which means it can't execute it, which means `simple` is not declared. But then your main snippet (in the html file) wants to use `simple`, and therefore blows up.

So the initial cause is that the .js cannot be fetched, which is a problem with the web server.

**Comment by chrismurrph** [comment]
Yes thanks. I'm guessing the direction for me to go if I ever need to fix that is to make spray routing do chunking to deal with the bigger 'fast' js file. Anyway the 'full' is up now. This picture shows both browsers. With Chrome I am able to move to another cell, with Firefox you cannot because it crashes.
![both_browsers](https://cloud.githubusercontent.com/assets/10278575/7965437/d3e61034-0a60-11e5-8ffa-2438df2e6845.png)

**Comment by sjrd** [comment]
I still have exactly the same errors as before here. Blank page because Firefox receives a truncated .js file, and Chrome says ERR_INVALID_CHUNKED_ENCODING.

**Comment by chrismurrph** [comment]
Thanks. Looks like I had better fix the ERR_INVALID_CHUNKED_ENCODING issue then. At least I will be able to see it myself when I run 'fast' on the remote server. 

The primary issue of this ticket is 'a' being undefined on Firefox. Let me know anything I can get to you now prior to me building a small Github demo as Li Haoyi suggested.

**Comment by sjrd** [comment]
So in fastOpt, the error is `element is undefined`. It refers to the parameter `element` of your method `simple.AtmosphereClient.Controller.clickTD`, which itself receives it from the `onclick` handler defined in a `def TD` somewhere in `simple.View`. This one extracts that value from `e.srcElement`, which is not a standard property. You should use `target` instead.

See http://stackoverflow.com/questions/5301643/how-can-i-make-event-srcelement-work-in-firefox-and-what-does-it-mean

In conclusion, this is not a Scala.js bug, so I'm going to close this issue.

Note also that you could probably have figured this out on your own if you developed in fastOpt mode in the first place, and used your browser's debugger (especially with source maps enabled).

**Comment by chrismurrph** [comment]
Thanks Sébastien

The next 'bug' I'll put on stackoverflow first. I'm now better at switching between fast/full opt. (I was returning to Scala.js when I discovered this apparent bug). I have not yet enabled source maps but will have another try sometime (I had problems last time I tried so gave up on them).

So I am trying to change my code to use target rather than srcElement. But in the code below the only thing I can get to compile is `e.srcElement`. `e` on its own and `e.event` don't compile. I'll put this up on stackoverflow and post back a reference from here when I get the answer.

```scala
  private def TD(locId: Location, gasId: Gas): HTMLTableCellElement = {
    val res = td(style := myGasesTable_td).render
    res.onclick = { (e: dom.Event) => c.clickTD(e.srcElement)}
    res.id = Id(locId, gasId).toString
    cells += (Id(locId, gasId) -> res)
    res
  }
```
